tests/src/item-namespace-staking.test.ts

```diff
@@ -0,0 +1,748 @@
+      new anchor.BN(itemIndex)
+    );
+
+    const args = {
+      classIndex: new anchor.BN(itemClassIndex),
+      parentClassIndex: null,
+      index: new anchor.BN(itemIndex),
+      stakingIndex: new anchor.BN(stakingIndex),
+      artifactClassMint: itemClassMintKeypair.publicKey,
+      artifactMint: itemMintKeypair.publicKey,
+      stakingPermissivenessToUse: { tokenHolder: true },
+    } as Instructions.Staking.BeginArtifactStakeCooldownArgs;
+
+    const accounts = {
+      artifactClass: artifactClassPDA,
+      artifact: artifactPDA,
+      stakingAccount: stakingMintAta,
+      parentClassAccount: null,
+      parentClassMint: null,
+      parentClass: null,
+      metadataUpdateAuthority: walletKeypair.publicKey,
+    } as Instructions.Staking.BeginArtifactStakeCooldownAccounts;
+
+    await stakingProgram.beginArtifactStakeCooldown(args, accounts);
+
+    const [stakingEscrow] =
+      await Utils.PDA.getArtifactIntermediaryStakingAccount({
+        artifactClassMint: itemClassMintKeypair.publicKey,
+        artifactMint: itemMintKeypair.publicKey,
+        index: new anchor.BN(itemIndex),
+        stakingMint: stakingMintKeypair.publicKey,
+        stakingIndex: new anchor.BN(stakingIndex),
+      });
+
+    const accInfo = await provider.connection.getTokenAccountBalance(
+      stakingEscrow,
+      "confirmed"
+    );
+
+    expect(accInfo.value.uiAmount).toBe(stakingAmount);
+  });
+
+  it("Ends artifact stake cooldown", async () => {
+    const stakingMintAta = await anchor.utils.token.associatedAddress({
+      mint: stakingMintKeypair.publicKey,
+      owner: walletKeypair.publicKey,
+    });
+
+    const [artifactClassPDA] = await Utils.PDA.getItemPDA(
+      itemClassMintKeypair.publicKey,
+      new anchor.BN(itemClassIndex)
+    );
+
+    const [artifactPDA] = await Utils.PDA.getItemPDA(
+      itemMintKeypair.publicKey,
+      new anchor.BN(itemIndex)
+    );
+
+    const args = {
+      classIndex: new anchor.BN(itemClassIndex),
+      index: new anchor.BN(itemIndex),
+      stakingIndex: new anchor.BN(stakingIndex),
+      artifactClassMint: itemClassMintKeypair.publicKey,
+      artifactMint: itemMintKeypair.publicKey,
+    } as Instructions.Staking.EndArtifactStakeCooldownArgs;
+
+    const accounts = {
+      artifactClass: artifactClassPDA,
+      artifact: artifactPDA,
+      stakingAccount: stakingMintAta,
+      stakingMint: stakingMintKeypair.publicKey,
+    } as Instructions.Staking.EndArtifactStakeCooldownAccounts;
+
+    await stakingProgram.endArtifactStakeCooldown(args, accounts);
+
+    const tokenAccounts =
+      await provider.connection.getParsedTokenAccountsByOwner(
+        walletKeypair.publicKey,
+        {
+          mint: stakingMintKeypair.publicKey,
+        },
+        "confirmed"
+      );
+
+    expect(tokenAccounts.value.length).toBe(1);
+
+    expect(
+      tokenAccounts.value[0].account.data.parsed.info.tokenAmount.uiAmount
+    ).toBe(stakingAmount);
+  });
+});
```

Comment:
We should add checks to ensure the artifact's tokens staked amount is being properly updated also.

Reply:
Actually, you probably only want to check the artifact's token amount and leave the current checks within the rust program tests.